### PR TITLE
Add aamtest support for Safari, implement AXAPI tests

### PR DESCRIPTION
**Pull request by twilco**
This commit registers the `aamtest` test type with `PytestExecutor` for Safari, and implements `test_axapi` for:
  - `role/button.py`
  - `role/button_haspopup.py`
  - `role/button_pressed.py`
  - `attribute/aria_braillelabel.py`
  - `attribute/aria_error_message.py`

### CODE REVIEW

File: core-aam/aamtests/attribute/aria_error_message.py [hunk 1]
```diff
@@ -18,11 +18,16 @@ def test_atspi(atspi, session, inline):
...
+def test_axapi(axapi, session, inline):
+    session.url = inline(TEST_HTML)
+
+    # Spec:
+    # Property: AXErrorMessageElements: pointers to accessible nodes matching IDREFs
+
+    node = axapi.find_node("test", session.url)
+    err, elements = axapi.AXUIElementCopyAttributeValue(node, "AXErrorMessageElements", None)
+    assert not err
+    assert elements and len(elements) == 1
```
Comment (jcsteh) [inline core-aam/aamtests/attribute/aria_error_message.py]: I think it might be good to test the element is actually the right element. To do that, we might need something in core-aam/aamtests/support/axapi_wrapper.py similar to atspi.get_relations_dictionary_helper which gets the elements and returns their DOM ids using AXDOMIdentifier.

File: core-aam/aamtests/role/button.py [hunk 2]
```diff
@@ -19,13 +19,18 @@ def test_atspi(atspi, session, inline, test_html):
...
+    session.url = inline(test_html)
+
+    # Spec:
+    # AXRole: AXButton
+    # AXSubrole: <nil>
+
+    node = axapi.find_node("test", session.url)
+    err, role = axapi.AXUIElementCopyAttributeValue(node, "AXRole", None)
+    assert not err
+    assert role == "AXButton"
```
Comment (jcsteh) [inline core-aam/aamtests/role/button.py:33]: Should we be asserting that AXSubrole is nil/non-existent?

**Review by jcsteh — CHANGES_REQUESTED** [review]
Thanks.

File: core-aam/aamtests/support/fixtures_a11y_api.py [hunk 3]
```diff
@@ -3,13 +3,15 @@
...
-    # TODO: add support for Edge, Safari.
+    # TODO: add support for Edge.
     if capabilities["browserName"] == "chrome":
         return capabilities["goog:processID"], "chrome"
     if capabilities["browserName"] == "firefox":
         return capabilities["moz:processID"], "firefox"
     if capabilities["browserName"] == "servo":
         return 0, "servo"
+    if "safari" in capabilities.get("browserName", "").lower():
+        return 0, capabilities["browserName"]
```
Comment (spectranaut) [inline core-aam/aamtests/support/fixtures_a11y_api.py]: Sorry about this! I fixed this same bug here: https://github.com/web-platform-tests/wpt/pull/60027, so you can rebase :)

File: core-aam/aamtests/attribute/aria_error_message.py (hunk 1 above)
Comment (twilco) [reply core-aam/aamtests/attribute/aria_error_message.py]: Good idea, thanks! Fixed in the latest commit.

File: core-aam/aamtests/role/button.py (hunk 2 above)
Comment (twilco) [reply core-aam/aamtests/role/button.py:33]: Definitely — fixed!

File: core-aam/aamtests/support/fixtures_a11y_api.py (hunk 3 above)
Comment (twilco) [reply core-aam/aamtests/support/fixtures_a11y_api.py]: No worries! In a follow-up PR I'll need to implement a proper `pid_from` in the same way Chrome and Firefox have, but this is OK for now.

**Review by jcsteh — APPROVED** [review]
Thank you.

File: core-aam/aamtests/attribute/aria_braillelabel.py [hunk 4]
```diff
@@ -11,11 +11,16 @@ def test_atspi(atspi, session, inline):
...
+def test_axapi(axapi, session, inline):
+    session.url = inline(TEST_HTML)
+
+    # Spec:
+    # AXBrailleLabel: <value>
+
+    node = axapi.find_node("test", session.url)
+    err, value = axapi.AXUIElementCopyAttributeValue(node, "AXBrailleLabel", None)
+    assert not err
+    assert value == "foobar"
```
Comment (cookiecrook) [inline core-aam/aamtests/attribute/aria_braillelabel.py:23]: ?

Comment (cookiecrook) [reply core-aam/aamtests/attribute/aria_braillelabel.py:23]: Nm. This looked like an error, but the prior test (hidden in the default view) really does use `aria-braillelabel='foobar'`

File: core-aam/aamtests/role/button.py [hunk 5]
```diff
@@ -19,13 +19,20 @@ def test_atspi(atspi, session, inline, test_html):
...
+    _, subrole = axapi.AXUIElementCopyAttributeValue(node, "AXSubrole", None)
```
Comment (cookiecrook) [inline core-aam/aamtests/role/button.py:34]: Probably could add another test (in another PR) that exercised a non-nil value for subrole, too.

Comment (cookiecrook) [reply core-aam/aamtests/role/button.py:34]: Actually that one is addressed below by AXToggle.

File: tools/wptrunner/wptrunner/browsers/safari.py [hunk 6]
```diff
@@ -24,7 +24,8 @@
                               "reftest": "WebDriverRefTestExecutor",
                               "wdspec": "PytestExecutor",
                               "crashtest": "WebDriverCrashtestExecutor",
-                              "test262": "WebDriverTestharnessExecutor"},
+                              "test262": "WebDriverTestharnessExecutor",
```
Comment (cookiecrook) [inline tools/wptrunner/wptrunner/browsers/safari.py:27]: TIL WebDriver and RefTest are studly capped but Testharness and Crashtest are not... Weird.

**Review by cookiecrook — APPROVED** [review]

**Comment by cookiecrook** [comment]
@gsnedders The CI check failures seems spurious. Are you able to merge this? Thanks.

**Comment by spectranaut** [comment]
@zcorpan can you merge this?

The firefox flake is unrelated, here is the bug for that: https://bugzilla.mozilla.org/show_bug.cgi?id=2035599

And the other one looks unrelated as well. Would be nice to get this in :)

**Review by spectranaut — APPROVED** [review]

**Review by harrisonwithrow-cloud — APPROVED** [review]

**Review by harrisonwithrow-cloud — APPROVED** [review]

**Review by harrisonwithrow-cloud — APPROVED** [review]

**Comment by jcscottiii** [comment]
Merging
